rest/src/main/kotlin/com/gitlab/kordlib/rest/Image.kt

```diff
@@ -18,19 +18,42 @@ class Image private constructor(val data: ByteArray, val format: Format) {
         }
 
     companion object {
-        fun raw(data: ByteArray, format: Format): Image {
-            return Image(data, format)
+        fun raw(data: ByteArray, format: Format, size: Size): Image {
+            return Image(data, format, size)
         }
 
         suspend fun fromUrl(client: HttpClient, url: String): Image = with(Dispatchers.IO) {
             val call = client.request<HttpResponse>(url) { method = HttpMethod.Get }
+            val size = Size.fromResolution(call.request.url.parameters["size"])
             val contentType = call.headers["Content-Type"]
-                    ?: error("expected 'Content-Type' header in image request")
+                ?: error("expected 'Content-Type' header in image request")
 
             @Suppress("EXPERIMENTAL_API_USAGE")
             val bytes = call.content.toByteArray()
 
-            Image(bytes, Format.fromContentType(contentType))
+            Image(bytes, Format.fromContentType(contentType), size)
+        }
+    }
+
+    /**
+     * Represents size of the Image as specified by
+     */
+    enum class Size(val resolution: String) {
+        SIZE_16("16"),
+        SIZE_32("32"),
+        SIZE_64("64"),
+        SIZE_128("128"),
+        SIZE_256("256"),
+        SIZE_512("512"),
+        SIZE_1024("1024"),
+        SIZE_2048("2048"),
+        SIZE_4096("4096");
+
+        companion object {
+            fun fromResolution(resolution: String?): Size {
+                return if (resolution == null) SIZE_128
+                else values().first { it.resolution == resolution }
+            }
```

Comment:
* Lack of documentation
* Why was 128 chosen as a default?

```kotlin
values().first { it.resolution == resolution }
```
This will throw an exception when `resolution` doesn't match any of the values.

Reply:
1. Sorry for missing docs, I'll update it.

2. When no resolution is specified in the url, discord sends us 128x128 image (or smaller if original res is smaller) by default. So to be explicit if resolution was not set in the request, we can put 128 by default.
*Note that, response resolution is always equal to or smaller than requested resolution i.e. if original res of image was 60x60 then even if we request 128, 256, or even 4096 we'll receive 60x60.*

Reply:
[discord.py sets default of 1024 :P](https://github.com/Rapptz/discord.py/blob/master/discord/user.py#L150), but as I observed (multiple avatars that were uploaded at different res) discord send us 128x128 by default so I just set it.

Ouch I still didn't removed this function (even if it is not used anywhere right now), sorry will do 👍